pkg/apis/picchu/v1alpha1/common.go

```diff
@@ -61,11 +61,41 @@ type PortInfo struct {
 	ContainerPort int32           `json:"containerPort,omitempty"`
 	Protocol      corev1.Protocol `json:"protocol,omitempty"`
 	Mode          PortMode        `json:"mode"`
-	HttpsRedirect bool            `json:"httpsRedirect,omitempty"`
 
 	Istio IstioPortConfig `json:"istio,omitempty"`
 }
 
+func (p *PortInfo) Merge(other *PortInfo) *PortInfo {
+	merged := p.DeepCopy()
+	SetPortDefaults(merged)
+	SetPortDefaults(other)
+	if other.Name != "" {
+		merged.Name = other.Name
+	}
+	if len(other.Hosts) > 0 {
+		merged.Hosts = other.Hosts
+	}
+	if other.IngressPort != defaultPortIngressPort {
```

Comment:
Would you also want a zero check?

Reply:
no, the Default call above ensures it's never zero